src/include/simeng/arch/aarch64/helpers/sve.hh

```diff
@@ -1019,14 +998,41 @@ class sveHelp {
     return {out, 256};
   }
 
+  /** Helper function for SVE instructions with the format `fmla zda, zn,
+   * zm[index]`.
+   * T represents the type of operands (e.g. for zn.d, T = uint64_t).
+   * Returns correctly formatted RegisterValue. */
+  template <typename T>
+  static RegisterValue sveMlaIndexed_vecs(
+      std::vector<RegisterValue>& operands,
+      const simeng::arch::aarch64::InstructionMetadata& metadata,
+      const uint16_t VL_bits) {
+    const T* d = operands[0].getAsVector<T>();
+    const T* n = operands[1].getAsVector<T>();
+    const T* m = operands[2].getAsVector<T>();
+    const size_t index = static_cast<size_t>(metadata.operands[2].vector_index);
+
+    const uint16_t elemsPer128 = 128 / (sizeof(T) * 8);
+    const uint16_t partition_num = VL_bits / (sizeof(T) * 8);
+    T out[256 / sizeof(T)] = {0};
```

Comment:
Yes. The size of any SVE instruction's RegisterValue is 256 bytes (the size it would be for a max VL of 2048-bits). Therefore the output array is pre-initialised to this maximum value.